forge-game/src/main/java/forge/game/spellability/SpellAbility.java

```diff
@@ -354,7 +354,7 @@ public int amountOfManaGenerated(boolean multiply) {
     public int totalAmountOfManaGenerated(SpellAbility saPaidFor, boolean multiply) {
         int result = 0;
         AbilityManaPart mp = getManaPart();
-        if (mp != null && metConditions() && mp.meetsManaRestrictions(saPaidFor)) {
```

Comment:
this is for chained Mana Effects like:
* Effect A: condition=false
* Effect B: condition=true

Reply:
the check is already inside `amountOfManaGenerated`
so it's redundant and only costs runtime

Reply:
but not in `InputMana::activateManaAbility`

Reply:
`activateManaAbility -> totalAmountOfManaGenerated -> amountOfManaGenerated -> metConditions`

nothing is skipped